### PR TITLE
[ext] Add dpsl types

### PR DESCRIPTION
**Pull request by zerefwayne**
To be merged after #20

### CODE REVIEW

File: diagnostics-extension/src/dpsl.types.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  modelName: string;
+}
+
+/**
+ * Response message containing CPU Info
+ */
+export interface CpuInfo {
+  numTotalThreads: number;
+  architecture: string;
+  physicalCpus: Array<unknown>; // Replace unknown
```
Comment (MahmoudAGawad) [inline diagnostics-extension/src/dpsl.types.ts]: is there more bound type than unknown?

File: diagnostics-extension/src/dpsl.types.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+/**
+ * Response message containing Fan Info
+ */
+export interface FanInfo {
+  speedRpm: number;
+}
```
Comment (MahmoudAGawad) [inline diagnostics-extension/src/dpsl.types.ts]: FanInfo should be an array.

https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/dpsl_types.js;drc=043c35d129e66b4ebe45509c4070a39a10fbd06f;l=80

File: diagnostics-extension/src/dpsl.types.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  | BacklightInfo
+  | BatteryInfo
+  | BluetoothInfo
+  | VpdInfo
+  | CpuInfo
+  | FanInfo
+  | MemoryInfo
+  | BlockDeviceInfo
+  | StatefulPartitionInfo
+  | TimezoneInfo;
```
Comment (MahmoudAGawad) [inline diagnostics-extension/src/dpsl.types.ts]: not sure if we need this. It was declared in dpsl/ for using inside the library code, not sure if the extension would need this?

File: diagnostics-extension/src/dpsl.types.ts [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  BLUEOOTH_ADAPTER_PROPERTY_CHANGED = 'bluetooth-adapter-property-changed',
+  BLUEOOTH_ADAPTER_REMOVED = 'bluetooth-adapter-removed',
+  BLUETOOTH_DEVICE_ADDED = 'bluetooth-device-added',
+  BLUETOOTH_DEVICE_PROPERTY_CHANGED = 'bluetooth-device-property-changed',
+  BLUETOOTH_DEVICE_REMOVED = 'bluetooth-device-removed',
+  LID_CLOSED = 'lid-closed',
+  LID_OPENED = 'lid-opened',
+  OS_RESUME = 'os-resume',
+  OS_SUSPEND = 'os-suspend',
+}
```
Comment (MahmoudAGawad) [inline diagnostics-extension/src/dpsl.types.ts]: These are internal definitions in dpsl. As a user of the library, you are going to call very specific event listeners, e.g. dpsl.system_events.bluetooth.addOnAdapterAddedListener(callback). More examples: https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/test/untrusted_browsertest.js?q=dpsl.system_events

Comment (zerefwayne) [reply diagnostics-extension/src/dpsl.types.ts]: >/**
     * Starts listening on bluetooth-adapter-added events.
     * @param {function()} callback
     * @public
     */
    addOnAdapterAddedListener(callback) {
      this.eventTarget.addEventListener(
          EVENTS.BLUETOOTH_ADAPTER_ADDED, callback);
    }
    
   Will I receive anything in the callback parameters? If yes, then where can I find their schemas?

Comment (MahmoudAGawad) [reply diagnostics-extension/src/dpsl.types.ts]: Callbacks[1] receive no input. It is just state change event, if you need more information, you can call the appropriate API. Example, you receive a callback for bluetooth-adapter-added, then you consult dpsl.telemetry.getBluetoothInfo() and get dpsl.BluetoothInfo[2] as a result.

[1] https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/events_service.js

[2] https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/dpsl_types.js;drc=043c35d129e66b4ebe45509c4070a39a10fbd06f;l=52

Comment (zerefwayne) [reply diagnostics-extension/src/dpsl.types.ts]: Makes sense. Thanks! 👍🏻 I've removed them from code.

**Review by MahmoudAGawad — APPROVED** [review]
LGTM. Thanks!

**Review by Lamzin — APPROVED** [review]

**Pull request merged**